common/src/main/java/org/apache/celeborn/common/write/PushState.java

```diff
@@ -105,4 +116,18 @@ public boolean limitZeroInFlight() throws IOException {
   public boolean reachLimit(String hostAndPushPort, int maxInFlight) throws IOException {
     return inFlightRequestTracker.reachLimit(hostAndPushPort, maxInFlight);
   }
+
+  public void startChecker() {
```

Comment:
We can add a flag inPushState indicating this object is in client side or in server side. If client, throw PUSH_DATA_TIMEOUT_MASTER in failExpiredBatch, else throw PUSH_DATA_TIMEOUT_SLAVE, thus we can delete PUSH_DATA_TIMEOUT

Reply:
Done